Userland/Libraries/LibGUI/TextEditor.cpp

```diff
@@ -181,7 +181,7 @@ TextPosition TextEditor::text_position_at_content_position(Gfx::IntPoint content
     switch (m_text_alignment) {
     case Gfx::TextAlignment::CenterLeft:
         for_each_visual_line(line_index, [&](Gfx::IntRect const& rect, auto& view, size_t start_of_line, [[maybe_unused]] bool is_last_visual_line) {
-            if (is_multi_line() && !rect.contains_vertically(position.y()) && !is_last_visual_line)
+            if (is_multi_line() && !rect.contains_vertically(position.y()) && !is_last_visual_line && position.y() >= 0)
```

Comment:
@konradekk I believe this was the reason the column index kept resetting to the line length after `y` got to values less than `0`. We needed to keep our `IterationDecision` set to continue only if `position.y` value is greater than or equal to 0.

When that condition wasn't there it just kept repeating the loop forever.